### PR TITLE
Handle null values in speed preset

### LINKED ISSUE #716: Create a speed optimized Preset
### Problem Description

After adding the framework in #715, we should add a configuration that allows the user to optimize a tabular model for speed.

### Expected behavior

A speed optimized tabular model will:
- Use the `GaussianCopula`
- Not round numerical data
- Enforce min/max values on numerical data
- Use the `CategoricalTransformer` (to be renamed: `FrequencyEncoder`) on any categorical columns
- Do not model null values. Instead, randomly generate null values based on observed proportions
- Default all univariate distributions to `gaussian` instead of performing a search

### API
The `TabularPreset` will accept 2 parameters:
- (required) `optimize_for`, the name of the preset. We'll add `"SPEED"`
- `metadata` a dictionary or `TableMetadata` object, similar to tabular models today

It will print out hard-coded info about the configuration & benchmarks.

```python
>>> from sdv.lite import TabularPreset

# the PresetModel will not allow too many extra settings, making it simple to use
>>> model = TabularPreset(optimize_for='SPEED', metadata=my_metadata)
Info: This config optimizes the modeling speed above all else.

Your exact runtime is dependent on the data. Benchmarks:
100K rows and 100 columns may take around 1 minute.
1M rows and 250 columns may take around 30 minutes.

# these work like any other tabular model
>>> model.fit(my_data)
>>> model.sample(my_data)
```

### Edge Cases
If the user does not pass in metadata, we should still run the model, but we should provide a warning:

```python
>>> from sdv.lite import TabularPreset
>>> model = TabularPreset(optimize_for='SPEED')
Warning: No metadata provided. Metadata will be automatically detected from your data. This process may
not be accurate. We recommend writing metadata to ensure correct data handling.

Info: This config optimizes the modeling speed above all else.

Your exact runtime is dependent on the data. Benchmarks:
100K rows and 100 columns may take around 1 minute.
1M rows and 250 columns may take around 30 minutes.
```

If the user does not pass in the required optimization parameter, throw an error.
```python
>>> from sdv.lite import TabularPreset
>>> model = TabularPreset()
Error: You must provide the name of a preset using the 'optimize_for' parameter. Use 
'TabularPreset.list_available_presets()' to browse through the options.
```

### PR DESCRIPTION
**Pull request by katxiao**
In the speed preset, we want to avoid modeling nulls in the GaussianCopula model. Instead, we capture the null value percentage of each column with null values before fitting, and add nulls back with the same likelihood after sampling. 

Resolves #716

### CODE REVIEW

File: sdv/lite/tabular.py [hunk 1]
```diff
@@ -42,22 +47,50 @@ def __init__(self, optimize_for=None, metadata=None):
...
 
     def sample(self, num_rows):
         """Sample rows from this table."""
-        return self._model.sample(num_rows)
+        sampled = self._model.sample(num_rows)
+
+        if self._null_percentages:
+            for field, percentage in self._null_percentages.items():
+                sampled[field] = sampled[field].apply(
+                    lambda x: np.nan if random.random() < percentage else x)
```
Comment (amontanez24) [inline sdv/lite/tabular.py:91]: I think `apply` can sometimes be slow. Can you try making a mask from the columns based on the null percentage for each one and applying the mask to the `DataFrame` instead? I think that should be faster. Here is an example https://cmdlinetips.com/2019/05/how-to-randomly-add-nan-to-pandas-dataframe/, although in this case the percentage of nulls is different for each field

**Review by amontanez24 — COMMENTED** [review]
This is looking really good! I just left one comment about potentially making it faster

File: sdv/lite/tabular.py (hunk 1 above)
Comment (katxiao) [reply sdv/lite/tabular.py:91]: Cool, sounds good! I'll update it

**Review by amontanez24 — APPROVED** [review]
LGTM! Thanks for addressing the changes
